internal/quickstart/choose_sdk.go

```diff
@@ -61,15 +62,90 @@ func NewChooseSDKModel(selectedIndex int) tea.Model {
 			Quit:          BindingQuit,
 		},
 		list:          l,
+		sdkDetails:    sdkDetails,
 		selectedIndex: selectedIndex,
 	}
 }
 
-// initSDKs sets the index of each SDK based on place in list.
-func initSDKs() {
-	for i := range SDKs {
-		SDKs[i].index = i
+// The CLI uses the sdkmeta project to obtain metadata about each SDK, including the display names
+// and types (client, server, etc.)
+// Currently, there is no sdkmeta for code examples associated with each SDK, so we hard-code the examples here.
+// Once they are part of sdkmeta we can remove this list.
+var sdkExamples = map[string]string{
+	"react-client-sdk":  "https://github.com/launchdarkly/react-client-sdk/tree/main/examples/typescript",
+	"vue":               "https://github.com/launchdarkly/vue-client-sdk/tree/main/example",
+	"react-native":      "https://github.com/launchdarkly/js-core/tree/main/packages/sdk/react-native/example",
+	"cpp-client-sdk":    "https://github.com/launchdarkly/cpp-sdks/tree/main/examples/hello-cpp-client",
+	"cpp-server-sdk":    "https://github.com/launchdarkly/cpp-sdks/tree/main/examples/hello-cpp-server",
+	"lua-server-server": "https://github.com/launchdarkly/lua-server-sdk/tree/main/examples/hello-lua-server",
+}
+
+// sdkOrder is a list of IDs the SDKs in order that they should be rendered based on popularity.
+var sdkOrder = []string{
+	"react-client-sdk",
+	"node-server",
+	"python-server-sdk",
+	"java-server-sdk",
+	"dotnet-server-sdk",
+	"js-client-sdk",
+	"vue",
+	"swift-client-sdk",
+	"go-server-sdk",
+	"android",
+	"react-native",
+	"ruby-server-sdk",
+	"flutter-client-sdk",
+	"dotnet-client-sdk",
+	"erlang-server-sdk",
+	"rust-server-sdk",
+	"cpp-client-sdk",
+	"roku",
+	"node-client-sdk",
+	"cpp-server-sdk",
+	"lua-server-sdk",
+	"haskell-server-sdk",
+	"php-server-sdk",
+}
+
+// initSDKs is responsible for loading SDK quickstart instructions from the embedded filesystem.
+//
+// The names of the files are special: they are the ID of the SDK (e.g. react-native), and are used as an index or
+// key to lookup associated sdk metadata (display name, SDK type, etc.)
+//
+// Therefore, take care when naming the files. A list of valid SDK IDs can be found here:
+// https://github.com/launchdarkly/sdk-meta/blob/main/products/names.json
+func initSDKs() []sdkDetail {
+	items, err := sdks.InstructionFiles.ReadDir("sdk_instructions")
+	if err != nil {
+		panic("failed to load embedded SDK quickstart instructions: " + err.Error())
 	}
+
+	slices.SortFunc(items, func(a fs.DirEntry, b fs.DirEntry) int {
+		return strings.Compare(a.Name(), b.Name())
+	})
+
+	details := make([]sdkDetail, 0, len(items))
+	for _, item := range items {
+		id, _, _ := strings.Cut(filepath.Base(item.Name()), ".")
+		if _, ok := sdkmeta.Names[id]; !ok {
+			continue
+		}
+		index := slices.Index(sdkOrder, id)
+		if index == -1 {
+			// if we missed an SDK don't add it with an invalid index
```

Comment:
Without this, there's a panic since it would later try to reference a -1 index. It _shouldn't_ happen unless we change the SDK ID and forget to update the order list.